DOCS-2557: Add architecture to platform reference page (#3251)
viamrobotics/docs: docs/platform.md

{{< board-carousel >}}
<br>

The open-source executable binary that runs on a Viam-powered {{< glossary_tooltip term_id="machine" text="machine" >}} is called `viam-server`.
`viam-server` runs and manages everything on your machine, including communications between hardware, software, and the cloud.

## What a machine is

A {{< glossary_tooltip term_id="machine" text="machine" >}} is any computer (SBC, server) running [`viam-server`](/architecture/#viam-server-and-the-micro-rdk) or any microcontroller running `viam-micro-server`, plus any hardware attached to that computer.

## How machines are structured, configured and organized

Learn more about the structure and configuration of machines:

|                        | Structure...                            | Configuration...                     |
| ---------------------- | --------------------------------------- | ------------------------------------ |
| ...of one machine ->   | [Architecture](/architecture/)          | [Machine Configuration](/configure/) |
| ...of many machines -> | [Cloud Organization Hierarchy](/cloud/) | [Deploy a Large Fleet](/fleet/)      |

## The things that make up a machine

> On your machine, you configure components and services.

Author: ```suggestion
On your machine, you configure [components](/components/) and [services](/services/).
```

Some are built-in, and many more are available in the [registry](/registry/).
You can also write your own and add them to the registry.

> On your machine, you can configure the following components:

Author: ```suggestion
You can configure the following components:
```

On your machine, you can configure the following components:
{{< cards >}}
{{% relatedcard link="/components/arm" %}}
{{% relatedcard link="/components/base" %}}
{{% relatedcard link="/components/board" %}}
{{% relatedcard link="/components/camera" %}}
{{% relatedcard link="/components/encoder" %}}
{{% relatedcard link="/components/gantry" %}}
{{% relatedcard link="/components/generic" %}}
{{% relatedcard link="/components/gripper" %}}
{{% relatedcard link="/components/input-controller" %}}
{{% relatedcard link="/components/motor" %}}
{{% relatedcard link="/components/movement-sensor" %}}
{{% relatedcard link="/components/power-sensor" %}}
{{% relatedcard link="/components/sensor" %}}
{{% relatedcard link="/components/servo" %}}
{{< /cards >}}
And you can use the following platform capabilities and services to make your machines smarter and better:
{{< cards >}}
{{% relatedcard link="/services/data" %}}
{{% relatedcard link="/services/ml/deploy" alt_title="Machine Learning" %}}
{{% relatedcard link="/services/motion" %}}
{{% relatedcard link="/services/navigation" %}}
{{% relatedcard link="/services/slam" %}}
{{% relatedcard link="/services/vision" %}}
{{< /cards >}}

> 

Author: ```suggestion
```

> {{< cards >}}

Reviewer: ```suggestion
```

> {{% card link="/registry/" %}}
> {{< /cards >}}

Reviewer: I think just the text link is fine. Given that all component pages will immediately also tell people about the registry I think that's probably enough
```suggestion
```

> {{< board-carousel >}}

Reviewer: thanks

<br>

The open-source executable binary that runs on a Viam-powered {{< glossary_tooltip term_id="machine" text="machine" >}} is called `viam-server`.

> `viam-server` runs and manages everything on your machine, including communications between hardware, software, and the cloud.

Author: I don't think it is, because this is saying what `viam-server` is whereas the next is saying what a machine is but not what `viam-server` is

> `viam-server` runs and manages everything on your machine, including communications between hardware, software, and the cloud.

Reviewer: hmm ok

> ## What a machine is in Viam

Author: Well, if I do "What is a machine in Viam" it becomes a question so I can do 
```suggestion
## What is a machine?
```
but I was not sure if we are allowing rhetorical questions

> ## What a machine is in Viam

Reviewer: either is fine.

> A machine is any computer (SBC, server, or microcontroller) running [`viam-server`](/architecture/#viam-server-and-the-micro-rdk) (or the micro-RDK in the case of a microcontroller), plus any hardware attached to that computer.

Author: ```suggestion
A {{< glossary_tooltip term_id="machine" text="machine" >}} is any computer (SBC, server) running [`viam-server`](/architecture/#viam-server-and-the-micro-rdk) or any microcontroller running `viam-micro-server`, plus any hardware attached to that computer.
```

## How machines are structured, configured and organized

Learn more about the structure and configuration of machines:

> |                        | Structure...                            | Configuration...                     |

Author: I had it that way initially but I personally liked top to bottom better?

> |                        | Structure...                            | Configuration...                     |

Reviewer: ok let's see when we test this on someone

| ---------------------- | --------------------------------------- | ------------------------------------ |
| ...of one machine ->   | [Architecture](/architecture/)          | [Machine Configuration](/configure/) |
| ...of many machines -> | [Cloud Organization Hierarchy](/cloud/) | [Deploy a Large Fleet](/fleet/)      |

{{< cards >}}
{{% card link="/architecture/" %}}
{{% card link="/configure/" %}}
{{% card link="/cloud/" %}}
{{% card link="/fleet/" %}}
{{< /cards >}}

## The things that make up a machine

On your machine, you configure components and services.
Some are built-in, and many more are available in the registry.
You can also write your own and add them to the registry.

> 
> In addition to modular components and services, the registry contains ML models.

Author: ```suggestion
```
